test/e2e/run.sh

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env bash
+
+set -o errexit
+
+declare -a on_exit_items
+
+function on_exit() {
+    if [ "${#on_exit_items[@]}" -gt 0 ]; then
+        echo -e '\nRunning deferred items, please do not interrupt until they are done:'
+    fi
+    for I in "${on_exit_items[@]}"; do
+        echo "deferred: ${I}"
+        eval "${I}"
+    done
+}
+
+# Cleaning up only makes sense in a local environment
+# it just wastes time in CircleCI 
+if [ "${CI}" != 'true' ]; then
+    trap on_exit EXIT
+fi
+
+function defer() {
+    on_exit_items=("$*" "${on_exit_items[@]}")
+}
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+SCRIPT_DIR="${REPO_ROOT}/test/e2e"
+KIND_VERSION=0.2.1
+CACHE_DIR="${REPO_ROOT}/cache/$CURRENT_OS_ARCH"
+KIND_CACHE_PATH="${CACHE_DIR}/kind-$KIND_VERSION"
+KIND_CLUSTER=flux-e2e
+USING_KIND=false
+FLUX_NAMESPACE=flux-e2e
+DEMO_NAMESPACE=demo
+
+
+# Check if there is a kubernetes cluster running, otherwise use Kind
+if ! kubectl version > /dev/null 2>&1 ; then
+    if [ ! -f "${KIND_CACHE_PATH}" ]; then
+        echo '>>> Downloading Kind'
+        mkdir -p "${CACHE_DIR}"
+        curl -sL "https://github.com/kubernetes-sigs/kind/releases/download/${KIND_VERSION}/kind-${CURRENT_OS_ARCH}" -o "${KIND_CACHE_PATH}"
+    fi
+    echo '>>> Creating Kind Kubernetes cluster'
+    cp "${KIND_CACHE_PATH}" "${REPO_ROOT}/test/bin/kind"
+    chmod +x "${REPO_ROOT}/test/bin/kind"
+    kind create cluster --name "${KIND_CLUSTER}" --wait 5m
+    defer kind --name "${KIND_CLUSTER}" delete cluster
+    export KUBECONFIG="$(kind --name="${KIND_CLUSTER}" get kubeconfig-path)"
+    USING_KIND=true
+    kubectl get pods --all-namespaces
+fi
+
+
+if ! helm version > /dev/null 2>&1; then
+    echo '>>> Installing Tiller'
+    kubectl --namespace kube-system create sa tiller
+    defer kubectl --namespace kube-system delete sa tiller
+    kubectl create clusterrolebinding tiller-cluster-rule --clusterrole=cluster-admin --serviceaccount=kube-system:tiller
+    defer kubectl delete clusterrolebinding tiller-cluster-rule
+    helm init --service-account tiller --upgrade --wait
+    defer helm reset --force
+fi
+
+kubectl create namespace "$FLUX_NAMESPACE"
+defer kubectl delete namespace "$FLUX_NAMESPACE"
+
+echo '>>> Installing mock git server'
+ssh-keygen -t rsa -N "" -f "${SCRIPT_DIR}/id_rsa"
+defer rm -f "${SCRIPT_DIR}/id_rsa" "${SCRIPT_DIR}/id_rsa.pub"
+kubectl create secret generic ssh-git --namespace="${FLUX_NAMESPACE}" --from-file="${SCRIPT_DIR}/known_hosts" --from-file="${SCRIPT_DIR}/id_rsa" --from-file=identity="${SCRIPT_DIR}/id_rsa" --from-file="${SCRIPT_DIR}/id_rsa.pub"
+kubectl apply -n "${FLUX_NAMESPACE}" -f "${SCRIPT_DIR}/gitsrv.yaml"
+kubectl -n "${FLUX_NAMESPACE}" rollout status deployment/gitsrv
+
+
+if [ "${USING_KIND}" = 'true' ]; then
+    echo '>>> Loading images into the Kind cluster'
```

Comment:
@squaremo @hiddeco Does your workflow with Minikube require something like this? 

From what I've read, people normally use the docker daemon in Minikube, but I wanted to double-check. In my setup (Docker for mac) I don't need to load any images since the k8s cluster uses the same docker daemon.

Reply:
I use the Docker daemon in Minikube (`eval $(minikube docker-env) && make`)

Reply:
Awesome, then this should work for you